scripts/generate-tracking-plan.ts

```diff
@@ -0,0 +1,326 @@
+      originalType = node;
+      return;
+    }
+
+    if (
+      ts.isTypeAliasDeclaration(node) &&
+      node.name.text === `Resolved${eventTypeName}`
+    ) {
+      targetResolvedType = node;
+    }
+  });
+
+  if (!originalType) {
+    throw new Error('cannot find originalType');
+  }
+
+  if (!targetResolvedType) {
+    throw new Error('cannot find targetResolvedType');
+  }
+
+  const originalSymbol = checker.getSymbolAtLocation(
+    (originalType as ts.TypeAliasDeclaration).name
+  );
+
+  const comment = originalSymbol
+    ? ts.displayPartsToString(originalSymbol.getDocumentationComment(checker))
+    : '';
+
+  const categoryTag = ts
+    .getJSDocTags(originalType)
+    .find((value: ts.JSDocTag) => {
+      return value.tagName.getText() === 'category';
+    });
+
+  return {
+    typeAlias: (originalType as ts.TypeAliasDeclaration)?.name.text,
+    category: categoryTag?.comment?.toString() ?? 'Other',
+    name: extractNamePropValue(targetResolvedType, checker),
+    comment: comment,
+    props: extractPayloadPropertiesAndComments(targetResolvedType, checker),
+  };
+}
+
+function getSourceWithReducedTypes(
+  originalSource: ts.SourceFile,
+  eventTypeNames: string[]
+) {
+  // Creates a new source file with new types for the events with a "squashed" payload,
+  // resolving any type reference to basic types.
+  // We then use the type checker to read the simplified types for each property.
+  // This allows us to write event types more freely, refactoring common interfaces, while
+  // being able to generate a readable tracking plan.
+
+  const modifiedSourceText = `
+type ResolveType<T> = T extends (...args: infer A) => infer R
+? (...args: ResolveType<A>) => ResolveType<R>
+: T extends object
+? T extends infer O
+  ? { [K in keyof O]: ResolveType<O[K]> }
+  : never
+: T;
+
+${originalSource.text}
+
+type ResolvedIdentifyTraits = {
+  name: 'Identify Traits',
+  payload: ResolveType<IdentifyTraits>
+};
+
+  ${eventTypeNames
+    .map((nodeName: string) => {
+      return `
+type Resolved${nodeName} = {
+  name: ${nodeName}['name'],
+  payload: ResolveType<${nodeName}['payload']>
+};
+    `;
+    })
+    .join('\n')}
+  `;
+
+  const sourceFile = ts.createSourceFile(
+    'inMemoryFile.ts',
+    modifiedSourceText,
+    ts.ScriptTarget.Latest,
+    true
+  );
+
+  const compilerOptions = {
+    // this is needed otherwise the type checker will remove undefined from any union
+    strictNullChecks: true,
+  };
+
+  const host = ts.createCompilerHost(compilerOptions);
+  host.getSourceFile = (fileName) =>
+    fileName === 'inMemoryFile.ts' ? sourceFile : undefined;
+
+  const program = ts.createProgram(['inMemoryFile.ts'], compilerOptions, host);
+  const checker = program.getTypeChecker();
+  return { sourceFile, checker };
+}
+
+function generateMarkdownPlan(
+  events: TelemetryEventInfo[],
+  identifyTraits: TelemetryEventInfo
+) {
+  const byCategory = _.groupBy(events, 'category');
```

Comment:
Totaly nit, since we're only using lodash for this can we do it in regular ts and avoid the dependency?

Reply:
True, removed and also sorted everything